### PR TITLE
task/VIDEO-10441/Implement_Unit_Tests_configuration

### PR DESCRIPTION
**Pull request by alejo740**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Pull Request Details

- Implemented Unit Testing configuration using JUnit and Espresso.
- Implemented some mock and fake classes in order to provide all the dependencies.
- Implemented test cases for SignIn Passcode Fragment.

### JIRA link(s):

- [VIDEO-10441](https://issues.corp.twilio.com/browse/VIDEO-10441)

### Description

In this PR all the Unit Testing environment is set with some basic test cases and ready to be used for future cases. Part of the classes were reused from the `Android-Video-Room-App`, but some conflicts were found due to now this `Live-Video-App` is fragment based.

### Before review
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with all effected platforms
* [ ] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Re-tested if necessary

### CODE REVIEW

File: apps/android/LiveVideoApp/app/src/debug/java/com/twilio/livevideo/app/ExtensionUtil.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import android.os.Bundle
+import androidx.annotation.StyleRes
+import androidx.core.util.Preconditions
+import androidx.fragment.app.Fragment
+import androidx.fragment.testing.R
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+
+@SuppressLint("RestrictedApi")
+inline fun <reified T : Fragment> launchFragmentInHiltContainer(
```
Comment (alejo740) [inline apps/android/LiveVideoApp/app/src/debug/java/com/twilio/livevideo/app/ExtensionUtil.kt:15]: This class is a workaround to make it `Hilt` testing works with fragments. By default, all the fragments requires a container (activity), but it's not provided by the framework.

File: apps/android/LiveVideoApp/app/src/debug/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (alejo740) [inline apps/android/LiveVideoApp/app/src/debug/AndroidManifest.xml:1]: `AndroidManifest.xml` only for debug and testing

File: apps/android/LiveVideoApp/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -11,7 +11,8 @@
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:supportsRtl="true"
-        tools:targetApi="31">
+        tools:targetApi="31"
+        android:networkSecurityConfig="@xml/network_security_config">
```
Comment (alejo740) [inline apps/android/LiveVideoApp/app/src/main/AndroidManifest.xml:15]: added new network security config to the main app, to allow external proxy intercept all the api Calls only on Debug Build.

**Review by timrozum — APPROVED** [review]

**Review by afalls-twilio — APPROVED** [review]
LGTM

**Pull request merged**